pkgs/by-name/sm/smc-fuzzer/package.nix

```
@@ -0,0 +1,39 @@
+{
+  lib,
+  swiftPackages,
+  fetchFromGitHub,
+  stdenv,
+}:
+
+# This isn’t just useful for fuzzing; it can also be used to explicitly
+# communicate with the SMC.  That makes it useful in system management,
+# e.g. querying / controlling charge status of an Apple laptop.
+stdenv.mkDerivation {
+  pname = "smc-fuzzer";
+  version = "20201223";
+
+  src = fetchFromGitHub {
+    repo = "smc-fuzzer";
+    owner = "theopolis";
+    rev = "ed60c4efeddea8a9778a976c49c650eee3840eac";
+    hash = "sha256-FyiFSVeO46UnBrpC8AhSuGe7alo37pT8J1qQWGPqV2U=";
+  };
+
+  buildInputs = [ swiftPackages.apple_sdk.frameworks.AppKit ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm 755 smc -t $out/bin
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Apple SMC (System Management Controller) API fuzzer";
+    homepage = "https://github.com/theopolis/smc-fuzzer";
```

```suggestion
    homepage = "https://github.com/theopolis/smc-fuzzer";
    license = lib.licenses.gpl2Plus;
```
https://github.com/theopolis/smc-fuzzer/blob/ed60c4efeddea8a9778a976c49c650eee3840eac/smc.h#L1-L19

---

Nice! I was looking for that. Thanks, I've also added a PR for a top-level LICENSE file upstream.